powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -206,12 +249,57 @@ public abstract class PowerAuthEncryptionProviderBase {
      */
     public @Nullable EciesEncryptedResponse encryptResponse(@Nonnull Object responseObject, @Nonnull PowerAuthEciesEncryption eciesEncryption) {
         try {
+            final EciesEncryptionContext encryptionContext = eciesEncryption.getContext();
+            final EciesScope eciesScope = encryptionContext.getEciesScope();
+
+            final String applicationKey = eciesEncryption.getContext().getApplicationKey();
+            final String ephemeralPublicKey = eciesEncryption.getContext().getEphemeralPublicKey();
+            final byte[] ephemeralPublicKeyBytes = Base64.getDecoder().decode(ephemeralPublicKey);
+
+            final PowerAuthEciesEncryptorParameters encryptorParameters;
+            // Obtain ECIES decryptor parameters from PowerAuth server
+            final byte[] associatedData;
+            final String version = eciesEncryption.getContext().getVersion();
+            final byte[] nonceBytesResponse = "3.2".equals(version) ? keyGenerator.generateRandomBytes(16) : null;
+            final String nonceResponse = Base64.getEncoder().encodeToString(nonceBytesResponse);
+            final Long timestampResponse = "3.2".equals(version) ? new Date().getTime() : null;
+            switch (eciesScope) {
+                case ACTIVATION_SCOPE -> {
+                    final String activationId = eciesEncryption.getContext().getActivationId();
+                    if (activationId == null) {
+                        logger.warn("Activation ID is required in ECIES activation scope");
+                        throw new PowerAuthEncryptionException();
+                    }
+                    encryptorParameters = getEciesEncryptorParameters(activationId, applicationKey, ephemeralPublicKey, encryptionContext.getVersion(), nonceResponse, timestampResponse);
```

Comment:
If I understand this correctly, we're getting the same key and sh2 from the PowerAuth Server again, right? The implementation of `getEciesEncryptorParameters` and `getEciesDecryptorParameters` does basically the same thing by calling PAS with using `powerAuthClient.getEciesDecryptor()`. 

I think this will have a noticeable performance impact. Is it possible to store such parameters in `PowerAuthEciesEncryption` and reuse it?

I don't understand the details but in the previous version we somehow managed to reuse the values from request decryption for the response encryption.

Reply:
Yes, this was my misunderstanding, we can still use `getEciesDecryptorParameters`. I will remove the new `getEciesEncryptorParameters` method.